xds/src/test/java/io/grpc/xds/CdsLoadBalancerTest.java

```diff
@@ -0,0 +1,289 @@
+
+    LoadBalancerRegistry lbRegistry = new LoadBalancerRegistry();
+    lbRegistry.register(new LoadBalancerProvider() {
+      @Override
+      public boolean isAvailable() {
+        return true;
+      }
+
+      @Override
+      public int getPriority() {
+        return 5;
+      }
+
+      @Override
+      public String getPolicyName() {
+        return "xds_experimental";
+      }
+
+      @Override
+      public LoadBalancer newLoadBalancer(Helper helper) {
+        edsLbHelpers.add(helper);
+        LoadBalancer edsLoadBalancer = mock(LoadBalancer.class);
+        edsLoadBalancers.add(edsLoadBalancer);
+        return edsLoadBalancer;
+      }
+    });
+
+    cdsLoadBalancer = new CdsLoadBalancer(helper, lbRegistry);
+  }
+
+  @Test
+  public void canHandleEmptyAddressListFromNameResolution() {
+    assertThat(cdsLoadBalancer.canHandleEmptyAddressListFromNameResolution()).isTrue();
+  }
+
+  @Test
+  public void missingCdsConfig() {
+    ResolvedAddresses resolvedAddresses = ResolvedAddresses.newBuilder()
+        .setAddresses(ImmutableList.<EquivalentAddressGroup>of())
+        .setAttributes(Attributes.newBuilder()
+            .set(XdsAttributes.XDS_CLIENT_REF, xdsClientRef).build())
+        .build();
+
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("Expecting a CDS LB config, but the actual LB config is 'null'");
+    cdsLoadBalancer.handleResolvedAddresses(resolvedAddresses);
+  }
+
+  @Test
+  public void invalidConfigType() {
+    Object invalidConfig = new Object();
+    ResolvedAddresses resolvedAddresses = ResolvedAddresses.newBuilder()
+        .setAddresses(ImmutableList.<EquivalentAddressGroup>of())
+        .setAttributes(Attributes.newBuilder()
+            .set(XdsAttributes.XDS_CLIENT_REF, xdsClientRef).build())
+        .setLoadBalancingPolicyConfig(invalidConfig)
+        .build();
+
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("Expecting a CDS LB config, but the actual LB config is '"
+        + invalidConfig + "'");
+    cdsLoadBalancer.handleResolvedAddresses(resolvedAddresses);
+  }
+
+  @Test
+  public void handleCdsConfigs() {
+    assertThat(xdsClient).isNull();
+
+    CdsConfig cdsConfig1 = new CdsConfig("foo.googleapis.com");
+    ResolvedAddresses resolvedAddresses1 = ResolvedAddresses.newBuilder()
+        .setAddresses(ImmutableList.<EquivalentAddressGroup>of())
+        .setAttributes(Attributes.newBuilder()
+            .set(XdsAttributes.XDS_CLIENT_REF, xdsClientRef).build())
+        .setLoadBalancingPolicyConfig(cdsConfig1)
+        .build();
+    cdsLoadBalancer.handleResolvedAddresses(resolvedAddresses1);
+
+    ArgumentCaptor<ClusterWatcher> clusterWatcherCaptor1 = ArgumentCaptor.forClass(null);
+    verify(xdsClient).watchClusterData(eq("foo.googleapis.com"), clusterWatcherCaptor1.capture());
+    assertThat(edsLbHelpers).hasSize(1);
+    assertThat(edsLoadBalancers).hasSize(1);
```

Comment:
Both of these checks should be after line 170, right? The call to `xdsClient.watchClusterData` didn't result into this?